### ISSUE TITLE
BLT strips the executable bit off my hooks

### ISSUE THREAD

**Issue by haynescw** [acquia/blt#361]
I tried to setup my first post code deploy hook. Im using jenkins to call blt's deploy command against our dev cloud account. I set the exectutable bit within our git repo, but after deploying our code to acquia's git server ten deploying the branch, i found that the executable bit was missing on my shell script.

**Comment by haynescw** [comment]
Im not entirely sure that this is just blt's doing, but a mixture of BLT with Jenkins. When i run the command on my local machine, the bit stays, when i run the command on my jenkins server, the bit gets lost.

**Comment by haynescw** [comment]
Im not entirely sure that this is just blt's doing, but a mixture of BLT with Jenkins. When i run the command on my local machine, the bit stays, when i run the command on my jenkins server, the bit gets lost.

**Comment by haynescw** [comment]
Im not entirely sure that this is just blt's doing, but a mixture of BLT with Jenkins. When i run the command on my local machine, the bit stays, when i run the command on my jenkins server, the bit gets lost.

**Comment by haynescw** [comment]
Im not entirely sure that this is just blt's doing, but a mixture of BLT with Jenkins. When i run the command on my local machine, the bit stays, when i run the command on my jenkins server, the bit gets lost.

**Comment by haynescw** [comment]
Im not entirely sure that this is just blt's doing, but a mixture of BLT with Jenkins. When i run the command on my local machine, the bit stays, when i run the command on my jenkins server, the bit gets lost.

**Comment by haynescw** [comment]
Im not entirely sure that this is just blt's doing, but a mixture of BLT with Jenkins. When i run the command on my local machine, the bit stays, when i run the command on my jenkins server, the bit gets lost.

**Comment by haynescw** [comment]
Im not entirely sure that this is just blt's doing, but a mixture of BLT with Jenkins. When i run the command on my local machine, the bit stays, when i run the command on my jenkins server, the bit gets lost.

**Comment by haynescw** [comment]
Im not entirely sure that this is just blt's doing, but a mixture of BLT with Jenkins. When i run the command on my local machine, the bit stays, when i run the command on my jenkins server, the bit gets lost.

**Comment by haynescw** [comment]
Im not entirely sure that this is just blt's doing, but a mixture of BLT with Jenkins. When i run the command on my local machine, the bit stays, when i run the command on my jenkins server, the bit gets lost.

**Comment by haynescw** [comment]
Im not entirely sure that this is just blt's doing, but a mixture of BLT with Jenkins. When i run the command on my local machine, the bit stays, when i run the command on my jenkins server, the bit gets lost.

**Comment by haynescw** [comment]
Im not entirely sure that this is just blt's doing, but a mixture of BLT with Jenkins. When i run the command on my local machine, the bit stays, when i run the command on my jenkins server, the bit gets lost.

**Comment by haynescw** [comment]
Im not entirely sure that this is just blt's doing, but a mixture of BLT with Jenkins. When i run the command on my local machine, the bit stays, when i run the command on my jenkins server, the bit gets lost.

**Comment by haynescw** [comment]
Im not entirely sure that this is just blt's doing, but a mixture of BLT with Jenkins. When i run the command on my local machine, the bit stays, when i run the command on my jenkins server, the bit gets lost.

**Comment by haynescw** [comment]
Im not entirely sure that this is just blt's doing, but a mixture of BLT with Jenkins. When i run the command on my local machine, the bit stays, when i run the command on my jenkins server, the bit gets lost.

**Comment by haynescw** [comment]
Im not entirely sure that this is just blt's doing, but a mixture of BLT with Jenkins. When i run the command on my local machine, the bit stays, when i run the command on my jenkins server, the bit gets lost.

**Comment by haynescw** [comment]
Im not entirely sure that this is just blt's doing, but a mixture of BLT with Jenkins. When i run the command on my local machine, the bit stays, when i run the command on my jenkins server, the bit gets lost.

**Comment by haynescw** [comment]
Im not entirely sure that this is just blt's doing, but a mixture of BLT with Jenkins. When i run the command on my local machine, the bit stays, when i run the command on my jenkins server, the bit gets lost.

**Comment by haynescw** [comment]
Im not entirely sure that this is just blt's doing, but a mixture of BLT with Jenkins. When i run the command on my local machine, the bit stays, when i run the command on my jenkins server, the bit gets lost.

**Comment by haynescw** [comment]
Im not entirely sure that this is just blt's doing, but a mixture of BLT with Jenkins. When i run the command on my local machine, the bit stays, when i run the command on my jenkins server, the bit gets lost.

**Comment by haynescw** [comment]
Im not entirely sure that this is just blt's doing, but a mixture of BLT with Jenkins. When i run the command on my local machine, the bit stays, when i run the command on my jenkins server, the bit gets lost.

**Comment by haynescw** [comment]
Im not entirely sure that this is just blt's doing, but a mixture of BLT with Jenkins. When i run the command on my local machine, the bit stays, when i run the command on my jenkins server, the bit gets lost.

**Comment by haynescw** [comment]
Im not entirely sure that this is just blt's doing, but a mixture of BLT with Jenkins. When i run the command on my local machine, the bit stays, when i run the command on my jenkins server, the bit gets lost.

**Comment by haynescw** [comment]
Im not entirely sure that this is just blt's doing, but a mixture of BLT with Jenkins. When i run the command on my local machine, the bit stays, when i run the command on my jenkins server, the bit gets lost.

**Comment by haynescw** [comment]
Im not entirely sure that this is just blt's doing, but a mixture of BLT with Jenkins. When i run the command on my local machine, the bit stays, when i run the command on my jenkins server, the bit gets lost.

**Comment by haynescw** [comment]
Im not entirely sure that this is just blt's doing, but a mixture of BLT with Jenkins. When i run the command on my local machine, the bit stays, when i run the command on my jenkins server, the bit gets lost.

**Comment by haynescw** [comment]
Im not entirely sure that this is just blt's doing, but a mixture of BLT with Jenkins. When i run the command on my local machine, the bit stays, when i run the command on my jenkins server, the bit gets lost.

**Comment by haynescw** [comment]
Im not entirely sure that this is just blt's doing, but a mixture of BLT with Jenkins. When i run the command on my local machine, the bit stays, when i run the command on my jenkins server, the bit gets lost.

**Comment by haynescw** [comment]
Im not entirely sure that this is just blt's doing, but a mixture of BLT with Jenkins. When i run the command on my local machine, the bit stays, when i run the command on my jenkins server, the bit gets lost.

**Comment by haynescw** [comment]
Im not entirely sure that this is just blt's doing, but a mixture of BLT with Jenkins. When i run the command on my local machine, the bit stays, when i run the command on my jenkins server, the bit gets lost.

**Comment by haynescw** [comment]
Im not entirely sure that this is just blt's doing, but a mixture of BLT with Jenkins. When i run the command on my local machine, the bit stays, when i run the command on my jenkins server, the bit gets lost.

**Comment by haynescw** [comment]
Im not entirely sure that this is just blt's doing, but a mixture of BLT with Jenkins. When i run the command on my local machine, the bit stays, when i run the command on my jenkins server, the bit gets lost.

**Comment by haynescw** [comment]
Im not entirely sure that this is just blt's doing, but a mixture of BLT with Jenkins. When i run the command on my local machine, the bit stays, when i run the command on my jenkins server, the bit gets lost.

**Comment by haynescw** [comment]
Im not entirely sure that this is just blt's doing, but a mixture of BLT with Jenkins. When i run the command on my local machine, the bit stays, when i run the command on my jenkins server, the bit gets lost.

**Comment by haynescw** [comment]
Im not entirely sure that this is just blt's doing, but a mixture of BLT with Jenkins. When i run the command on my local machine, the bit stays, when i run the command on my jenkins server, the bit gets lost.

**Comment by haynescw** [comment]
Im not entirely sure that this is just blt's doing, but a mixture of BLT with Jenkins. When i run the command on my local machine, the bit stays, when i run the command on my jenkins server, the bit gets lost.

**Comment by haynescw** [comment]
Im not entirely sure that this is just blt's doing, but a mixture of BLT with Jenkins. When i run the command on my local machine, the bit stays, when i run the command on my jenkins server, the bit gets lost.

**Comment by haynescw** [comment]
Im not entirely sure that this is just blt's doing, but a mixture of BLT with Jenkins. When i run the command on my local machine, the bit stays, when i run the command on my jenkins server, the bit gets lost.

**Comment by grasmash** [comment]
I recall a similar issue being resolved in the last couple of months, how recent is your BLT version?

**Comment by haynescw** [comment]
Is there a command to get the current version or are you just wanting to make sure that i pulled down and copied over the most recent version of blt's scripts?

**Comment by haynescw** [comment]
I ended up just adding a chmod command inside the the deploy:copy phing command after this line https://github.com/acquia/blt/blob/8.x/phing/tasks/deploy.xml#L123

and that solved my issue.
Are those  files supposed to be updated when you do a blt:update command?

**Comment by grasmash** [comment]
The rsync command [here](https://github.com/acquia/blt/blob/8.x/phing/tasks/deploy.xml#L116) uses a `-a` which should cause permissions to be preserved for the copied files, including `hooks`.

Can you run `blt deploy:build` locally and then check to see the permissions on `deploy/hooks`? That should let us know if it's blt or Jenkins that's losing the bit.

**Comment by haynescw** [comment]
Here's everything I did before i explicitly made the change.

I ran deploy:build on my mac to find that the command ran completely fine.
Then I ran it on my jenkins server as the jenkins user and thats where I found that it wasnt working.
So i figured it had to be specific to the jenkins server. So I updated our project to make it work.

That being said, our jenkins server is in our azure cloud and was build from scratch . Im using ubuntu 16.04 and have only the software needed to run our build for our current project. 

I guess this seems like an anomaly. I will continue to debug why this is happening on our jenkins server to figure out if just the rsync command can be ran without phing. I have a hunch this might be an app armor thing trying to protect our jenkins instance.
